arangod/Replication2/ReplicatedLog/LogFollower.cpp

```diff
@@ -449,16 +473,17 @@ replicated_log::LogFollower::LogFollower(
 
 auto replicated_log::LogFollower::waitFor(LogIndex idx)
     -> replicated_log::ILogParticipant::WaitForFuture {
-  auto self = _guardedFollowerData.getLockedGuard();
+  auto self = _guardedFollowerData.getLockedGuard(ADB_HERE);
   if (self->_commitIndex >= idx) {
     return futures::Future<WaitForResult>{
         std::in_place, self->_commitIndex,
         std::make_shared<QuorumData>(idx, _currentTerm)};
   }
   // emplace might throw a std::bad_alloc but the remainder is noexcept
   // so either you inserted it and or nothing happens
-  auto it =
-      self->_waitForQueue.getLockedGuard()->emplace(idx, WaitForPromise{});
+  // TODO locking ok? Iterator stored but lock guard is temporary
```

Comment:
Let's add a ticket for that, so we don't forget.

Reply:
https://arangodb.atlassian.net/browse/CINFRA-338